Based on: https://tools.suckless.org/dmenu/patches/lines-below-prompt/

By me: Added parameter to enable this on demand
---
diff --git a/dmenu.c b/dmenu.c
index 17256e8..fe899b4 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -53,6 +53,7 @@ static struct item *items = NULL;
 static struct item *matches, *matchend;
 static struct item *prev, *curr, *next, *sel;
 static int mon = -1, screen;
+static int lines_below_prompt = 0;
 
 static Atom clip, utf8;
 static Display *dpy;
@@ -162,8 +163,13 @@ drawmenu(void)
 
 	if (lines > 0) {
 		/* draw vertical list */
-		for (item = curr; item != next; item = item->right)
-			drawitem(item, x, y += bh, mw - x);
+		for (item = curr; item != next; item = item->right) {
+			if (lines_below_prompt) {
+				drawitem(item, x - promptw, y += bh, mw);
+			} else {
+				drawitem(item, x, y += bh, mw - x);
+			}
+		}
 	} else if (matches) {
 		/* draw horizontal list */
 		x += inputw;
@@ -798,6 +804,7 @@ usage(void)
 {
 	fputs("usage: dmenu [-bfiv] [-l lines] [-h height] [-p prompt] [-fn font] [-m monitor]\n"
 	      "             [-x xoffset] [-y yoffset] [-z width]\n"
+	      "             [-lbp]\n"
 	      "             [-nb color] [-nf color] [-sb color] [-sf color] [-w windowid]\n"
 	      "             [-d separator] [-D separator]\n", stderr);
 	exit(1);
@@ -820,6 +827,8 @@ main(int argc, char *argv[])
 			fast = 1;
 		else if (!strcmp(argv[i], "-F"))   /* grabs keyboard before reading stdin */
 			fuzzy = 0;
+		else if (!strcmp(argv[i], "-lbp"))  /* enable lines lines_below_prompt */
+			lines_below_prompt = 1;
 		else if (!strcmp(argv[i], "-i")) { /* case-insensitive item matching */
 			fstrncmp = strncasecmp;
 			fstrstr = cistrstr;
